src/engine/strat_engine/backstore/util.rs

```diff
@@ -72,7 +82,14 @@ pub fn get_stratis_block_devices() -> StratisResult<Vec<PathBuf>> {
 
     let devices: Vec<PathBuf> = enumerator
         .scan_devices()?
-        .map(|x| x.devnode().expect("block devices have devnode").into())
+        .filter(|dev| {
+            dev.property_value("DM_MULTIPATH_DEVICE_PATH").is_none() && dev.devnode().is_some()
+        })
+        .map(|x| {
+            x.devnode()
+                .expect("we filtered out missing devnode entries")
+                .into()
+        })
```

Comment:
for both `get_stratis_block_devices` and `get_all_empty_devices` instead of filtering on `dev.devnode.is_some()` and then doing a `map()` with `expect()`, you can instead do a filter_map:
```
     .filter_map(|i| i.devnode().map(|d| d.into()))
```
This prevents the unfortunate situation where I didn't immediately see the filter was filtering on that (it's all the way over on the right of the filter line (c'mon rustfmt, step it up!)) , and also puts all the devnode stuff, both the filtering and the mapping, together.